test/Microsoft.AspNetCore.Razor.Tools.Test/Infrastructure/ToolTestFixture.cs

```diff
@@ -0,0 +1,124 @@
+{
+    public class ToolTestFixture : IDisposable
+    {
+        private readonly string _defaultNugetPackageLocation;
+
+        public ToolTestFixture()
+        {
+            PackSrcDirectory();
+
+            _defaultNugetPackageLocation = Environment.GetEnvironmentVariable("NUGET_PACKAGES");
+            if (string.IsNullOrWhiteSpace(_defaultNugetPackageLocation))
+            {
+                _defaultNugetPackageLocation = Environment.ExpandEnvironmentVariables(
+                    @"%USERPROFILE%\.nuget\packages");
+            }
+
+            Environment.SetEnvironmentVariable("NUGET_PACKAGES", TestArtifactsTempPackagesDirectory);
+
+            InitializeTestApps();
+        }
+
+        public string RootDirectory => Path.GetFullPath(Path.Combine("..", ".."));
+
+        public string SrcDiretory => Path.Combine(RootDirectory, "src");
+
+        public string TestArtifactsDirectory => Path.Combine(RootDirectory, "testartifacts");
+
+        public string TestArtifactsTempPackagesDirectory => Path.Combine(TestArtifactsDirectory, "packages");
+
+        public string TestAppDirectory => Path.Combine(RootDirectory, "testapps");
+
+        public StringBuilder DotNet(string projectDirectory, string commandName, params string[] args)
+        {
+            var command = Command.CreateDotNet(commandName, args);
+            var outputWriter = new StringWriter();
+
+            var exitCode = command
+                .ForwardStdErr(outputWriter)
+                .ForwardStdOut(outputWriter)
+                .WorkingDirectory(projectDirectory)
+                .Execute()
+                .ExitCode;
+
+            var outputBuilder = outputWriter.GetStringBuilder();
+
+            if (exitCode != 0)
+            {
+                Assert.True(false, outputBuilder.ToString());
+            }
+
+            return outputBuilder;
+        }
+
+        public void Dispose()
+        {
+            Environment.SetEnvironmentVariable("NUGET_PACKAGES", _defaultNugetPackageLocation);
+            Directory.Delete(TestArtifactsTempPackagesDirectory, recursive: true);
+        }
+
+        private void PackSrcDirectory()
+        {
+            Console.WriteLine("Packing src projects...");
+
+            DotNet(SrcDiretory, "restore");
+
+            foreach (var directory in Directory.EnumerateDirectories(SrcDiretory))
+            {
+                DotNet(directory, "pack", "-o", TestArtifactsDirectory);
+            }
+
+            Console.WriteLine("Done.");
+        }
+
+        private void InitializeTestApps()
+        {
+            var outputWriter = new StringWriter();
+            var outputBuilder = outputWriter.GetStringBuilder();
+
+            if (Directory.Exists(TestArtifactsTempPackagesDirectory))
+            {
+                Directory.Delete(TestArtifactsTempPackagesDirectory, recursive: true);
+            }
+
+            var nugetConfigXmlLocation = Path.Combine(RootDirectory, "NuGet.config");
```

Comment:
This may not always work. Build CI can change/redirect nuget feeds in a few different ways.